core/src/main/java/io/crate/plugin/PluginLoader.java

```diff
@@ -132,36 +115,31 @@ public String apply(Plugin input) {
         }
     }
 
-    private Set<Class<? extends Plugin>> removeAndLogDuplicates(List<Class<? extends Plugin>> allImplementations) {
-        Set<Class<? extends Plugin>> uniques = Sets.newHashSet();
-        for(Class<? extends Plugin> elem : allImplementations) {
-            if (uniques.contains(elem)) {
-                CodeSource codeSource = elem.getProtectionDomain().getCodeSource();
-                String pluginStr = (codeSource != null ? codeSource.getLocation().toString() : elem.getName());
-                logger.warn("Plugin [{}] already loaded. ignoring...", pluginStr);
-            }
-            uniques.add(elem);
-        }
-        return uniques;
-    }
-
-    @Nullable
-    private List<URL> getPluginUrls() {
-        if (!isAccessibleDirectory(pluginPath, logger)) {
+    private Collection<Class<? extends Plugin>> findImplementations() {
+        if (!isAccessibleDirectory(pluginsPath, logger)) {
             return Collections.emptyList();
         }
 
-        File[] plugins = pluginPath.toFile().listFiles();
+        File[] plugins = pluginsPath.toFile().listFiles();
         if (plugins == null) {
             return Collections.emptyList();
         }
 
-        List<URL> pluginUrls = new ArrayList<>();
+        Collection<Class<? extends Plugin>> allImplementations = new ArrayList<>();
         for (File plugin : plugins) {
             if (!plugin.canRead()) {
                 logger.debug("[{}] is not readable.", plugin.getAbsolutePath());
                 continue;
             }
+            // check if its an elasticsearch plugin
+            try {
+                PluginInfo.readFromProperties(plugin.toPath());
```

Comment:
wouldn't it be enough to just check for the descriptor file instead of parsing the whole file?